### PR TITLE
[LLC] Refine generated output

### PR DESCRIPTION
**Pull request by chamons**
- Part of https://github.com/Azure/autorest.csharp/issues/1104

## Improvements
- Remove generated Enums in Models folder, falling back to the base type instead in APIs that would use them
- Pipeline is now protected field
- "body" param is now "requestBody"
- Create*Request method is now protected
- Added test sample.

**Additional Refactor:** Hoist Model code from OutputLibrary.cs to DataPlaneOutputLibrary.cs since Models are not a universal concept

### CODE REVIEW

File: src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs [hunk 1]
```diff
@@ -115,7 +114,7 @@ private void WriteClientMethod(CodeWriter writer, RestClientMethod clientMethod,
 
         private void WriteClientFields(CodeWriter writer, LowLevelRestClient client, BuildContext context)
         {
-            writer.Line($"private readonly {typeof(HttpPipeline)} {PipelineField};");
+            writer.Line($"protected readonly {typeof(HttpPipeline)} {PipelineField};");
```
Comment (pakrym) [inline src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:117]: Should this be a property?

Comment (chamons) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:117]: Yes, I think so. Let me give it a swing.

Comment (chamons) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs]: Done.

**Comment by chamons** [comment]
Darn, messed up the merge. One sec.

**Comment by pakrym** [comment]
We should really add some LLC-generated code to at least illustrate changes and make sure it builds. I'm OK if it's a sample instead of tests for now.

**Review by pakrym — APPROVED** [review]

File: src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs [hunk 2]
```diff
@@ -115,7 +114,8 @@ private void WriteClientMethod(CodeWriter writer, RestClientMethod clientMethod,
 
         private void WriteClientFields(CodeWriter writer, LowLevelRestClient client, BuildContext context)
         {
-            writer.Line($"private readonly {typeof(HttpPipeline)} {PipelineField};");
+            writer.Append($"protected {typeof(HttpPipeline)} {PipelineField}");
+            writer.AppendRaw("{ get; }\n");
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:118]: Can we add a test project to see these changes and add some tests for it?

Comment (chamons) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:118]: I'm trying to wait for LLC output to be stabilized before adding a test project. 

Maybe I should give in and add one, thoughts @pakrym ?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs (hunk 2 above)
Comment (pakrym) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:118]: https://github.com/Azure/autorest.csharp/pull/1109#issuecomment-812012866 😄

File: src/AutoRest.CSharp/Output/Models/Types/MgmtOutputLibrary.cs [hunk 3]
```diff
@@ -175,18 +179,30 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
...
             if (!SchemaMap.TryGetValue(schema, out result) && !ResourceSchemaMap.TryGetValue(schema, out result))
             {
                 throw new KeyNotFoundException($"{schema.Name} was not found in model and resource schema map");
             }
 
-            return result;
+            return result.Type;
+        }
+
+        public override CSharpType? FindTypeByName(string name)
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/AutoRest.CSharp/Output/Models/Types/MgmtOutputLibrary.cs:193]: What do you mean by "original name"?

File: src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs (hunk 2 above)
Comment (ShivangiReja) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:118]: I didn't mean to add a test project in `autorest.testserver` repo. I meant we can add test project [here](https://github.com/Azure/autorest.csharp/tree/feature/v3/test/TestProjects), if you see here we have only c# related test projects. We can add/update projects anytime here as per our needs.

The plus side of adding a project and tests would to see the differences made by the PR and see if the changes broke any tests.

File: src/AutoRest.CSharp/Output/Models/Types/MgmtOutputLibrary.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (chamons) [reply src/AutoRest.CSharp/Output/Models/Types/MgmtOutputLibrary.cs]: I renamed it, but I agree I don't understand the difference.

File: src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs (hunk 2 above)
Comment (chamons) [reply src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs:118]: I added a top level sample for this PR.

As discussed in the call, detailed swaggers are coming later. I've file https://github.com/Azure/autorest.csharp/issues/1115 to make sure that work doesn't get lost.

File: samples/Azure.AI.DocumentTranslation/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Azure.AI.DocumentTranslation
+
+### AutoRest Configuration
+
+> see https://aka.ms/autorest
+
+```yaml
+title: Azure.AI.DocumentTranslation
+tag: release_1_0_preview.1
```
Comment (ShivangiReja) [inline samples/Azure.AI.DocumentTranslation/readme.md:9]: nit: tag is not needed here

**Comment by chamons** [comment]
@m-nash Issues resolved. Please re-review and look at the question comment.

**Review by ShivangiReja — APPROVED** [review]

**Review by m-nash — APPROVED** [review]

**Pull request merged**